src/runtime/virtcontainers/clh.go

```diff
@@ -670,7 +672,6 @@ func (clh *cloudHypervisor) hotplugAddBlockDevice(drive *config.BlockDrive) erro
 	clhDisk := *chclient.NewDiskConfig(drive.File)
 	clhDisk.Readonly = &drive.ReadOnly
 	clhDisk.VhostUser = func(b bool) *bool { return &b }(false)
-	clhDisk.Id = &driveID
 
```

Comment:
Just out of curiosity (and my apologies if the following questions are stupid as I'm totally not familiar with this code):

- we're now leaving `clhDisk.Id` uninitialised, is that OK?  From my cursory check the surrounding code seems prepared in general to handle an empty `clhDisk.Id`, just asking if there might be a code path that relies on a non-empty value for case of a hot-plugged drive.
- would it be possible to store the new device ID right in `clhDisk.Id` instead of a newly added map?

Reply:
@pmores, great question.

The clhDisk.Id is only used for raising a request to Cloud Hypervisor, and then, **if** filled, Cloud Hypervisor will use the id passed to manage the device internally using that Id.  If we do **not** pass anything, Cloud Hypervisor will generate it automatically, and that's the thing we're taking advantage here.

Storing it clhDisk is **not** the place to store it, as it's ephemeral, it's just a variable created to raise a request for Cloud Hypervisor to add a disk on its side.  Once this function is over, that variable is out of scope and doesn't exist anymore.  That's the reason we store the Cloud Hypervisor's generated id in a map, and then use that to match the drive / device ID coming from the sandbox layer.

The question then is ... can we change the `drive.ID`?  And the answer is not without a really good investigation and a deep dive into what's done inside other VMMs, and that's something I'm not willing to risk while solving a bug that close to the release.

Did I answer all your questions, @pmores?

Reply:
@fidencio That makes perfect sense, thanks a lot for the explanation. I have no further questions - would you like me also to approve the PR formally?

Reply:
If you're comfortable doing so, I'd appreciate that. :-)